integrationTests/testInitializer.go

```diff
@@ -169,15 +173,19 @@ func CreateMetaStore(coordinator sharding.Coordinator) dataRetriever.StorageServ
 }
 
 // CreateAccountsDB creates an account state with a valid trie implementation but with a memory storage
-func CreateAccountsDB(shardCoordinator sharding.Coordinator) *state.AccountsDB {
-	hasher := sha256.Sha256{}
-	store := CreateMemUnit()
+func CreateAccountsDB(shardCoordinator sharding.Coordinator,
+) (*state.AccountsDB, data.Trie, storage.Storer) {
 
-	tr, _ := trie.NewTrie(store, TestMarshalizer, hasher)
 	accountFactory, _ := factory.NewAccountFactoryCreator(shardCoordinator)
-	adb, _ := state.NewAccountsDB(tr, sha256.Sha256{}, TestMarshalizer, accountFactory)
+	if shardCoordinator == nil {
```

Comment:
I would do the shardCoordinator == nil check before accountFactory creation, otherwise it looks weird.

Reply:
Replaced with an if-else statement